s3/src/test/scala/com/meltwater/fawn/s3/S3PayloadsSpec.scala

```diff
@@ -0,0 +1,281 @@
+package com.meltwater.fawn.s3
+
+import cats.effect.IO
+import net.andimiller.munit.cats.effect.styles.FlatIOSpec
+import org.http4s.scalaxml.xml
+import org.http4s.{EntityDecoder, Header, Response}
+
+import scala.xml.Elem
+
+class S3PayloadsSpec extends FlatIOSpec {
+
+  val listBucketsExample: String =
+    """<?xml version="1.0" encoding="UTF-8"?>
+      |<ListAllMyBucketsResult>
+      | <Buckets>
+      |  <Bucket>
+      |    <CreationDate>2022-01-17</CreationDate>
+      |    <Name>bucket-name</Name>
+      |  </Bucket>
+      | </Buckets>
+      | <Owner>
+      | <DisplayName>accountname</DisplayName>
+      | <ID>accountid</ID>
+      | </Owner>
+      |</ListAllMyBucketsResult>""".stripMargin
+
+  test("decode a list all my buckets response") {
+    EntityDecoder[IO, Elem]
+      .decode(
+        Response[IO]()
+          .withEntity(listBucketsExample)
+          .withHeaders(Header("content-type", "text/xml")),
```

Comment:
if you defined `listBucketsExample` as an XML literal of type `Elem` this would be set for you by `withEntity`